Lib/test/test_importlib/fixtures.py

```diff
@@ -0,0 +1,196 @@
+from __future__ import unicode_literals
+
+import os
+import sys
+import shutil
+import tempfile
+import textwrap
+import contextlib
+
+try:
+    from contextlib import ExitStack
+except ImportError:
+    from contextlib2 import ExitStack
+
+try:
+    import pathlib
+except ImportError:
+    import pathlib2 as pathlib
+
+
+__metaclass__ = type
+
+
+@contextlib.contextmanager
+def tempdir():
+    tmpdir = tempfile.mkdtemp()
+    try:
+        yield pathlib.Path(tmpdir)
+    finally:
+        shutil.rmtree(tmpdir)
+
+
+@contextlib.contextmanager
+def save_cwd():
+    orig = os.getcwd()
+    try:
+        yield
+    finally:
+        os.chdir(orig)
+
+
+@contextlib.contextmanager
+def tempdir_as_cwd():
```

Comment:
Is this same as `support.temp_cwd` at https://github.com/python/cpython/blob/6236c9823ef3e8e2229b0598d3d8189adf5e00f2/Lib/test/support/__init__.py#L1042 ?

Reply:
Yes, very similar.

Reply:
I don't think it can use `test.support.temp_cwd` in part because it doesn't call `test.test_importlib.fixtures.tempdir()`, which has a different interface (returning a `pathlib.Path`). Since the upstream project doesn't have these constructs, I'd prefer to stick with the local implementations until this implementation becomes canonical and `importlib_metadata` becomes the backport (planned after the 3.8 release).